views/newsletters.html

```diff
@@ -337,4 +340,25 @@
   });
 </script>
 <% endif %>
+
+<script>
```

Comment:
Good work encapsulating the logic! I incorporated your code with two fixes. You now use component state, but you still have to persist it in global variables, otherwise it will show the red dot and reload Beamer JS again when switching between tabs.